tests/cpp/test_multidevice_lower_communication.cpp

```diff
@@ -536,7 +503,158 @@ TEST_P(LowerCollectiveTest, ReduceScatter_Allgather) {
   EXPECT_TRUE(at::allclose(out_tensor, unsharded_in_tensor.sum(0)));
 }
 
-TEST_P(LowerCollectiveTest, AllgatherLoopSplit_Noncontig) {
+INSTANTIATE_TEST_SUITE_P(
+    HostIrLowering,
+    LowerCollectiveTest,
+    ::testing::Combine(
+        testing::Values(CommunicatorBackend::kNccl, CommunicatorBackend::kUcc),
+        testing::Values(false)),
+    ([](const testing::TestParamInfo<std::tuple<CommunicatorBackend, bool>>&
+            info) -> std::string {
+      const auto& [backend_type, enable_host_ir_lowering] = info.param;
+      std::stringstream ss;
+      ss << backend_type;
+      ss
+          << (enable_host_ir_lowering ? "_HirLowerEnabled"
+                                      : "_HirLowerDisabled");
+      return ss.str();
+    }));
+
+class LowerNoncontigCollectiveTest
+    : public MultiDeviceTest,
+      public testing::WithParamInterface<
+          std::tuple<CommunicatorBackend, bool, bool, bool>> {
+  // params = [backend, input_contiguous, input_did_outermost, output_allocated]
+  // backend: the backend type to use.
+  // input_contiguous_: If input tensorview is contiguous.
+  // input_did_outermost_: If the input tensorview's sharded axis is outermost.
+  // output_allocated_: If the output tensorview has an allocation.
+ protected:
+  LowerNoncontigCollectiveTest() : optimization_guard_(false) {}
```

Comment:
Why? While I agree on the end state where MAPP will run after ReorderedShardedAxis, MAPP runs before. What issues does this optimization guard try to hide?

Reply:
Recording offline discussion:
We will add the guard for now, and remove it after https://github.com/NVIDIA/Fuser/issues/4566